lib/connections/quic_listen.go

```diff
@@ -82,18 +82,24 @@ func (t *quicListener) OnExternalAddressChanged(address *stun.Host, via string)
 func (t *quicListener) serve(ctx context.Context) error {
 	network := strings.ReplaceAll(t.uri.Scheme, "quic", "udp")
 
-	packetConn, err := net.ListenPacket(network, t.uri.Host)
+	udpAddr, err := net.ResolveUDPAddr(network, t.uri.Host)
 	if err != nil {
 		l.Infoln("Listen (BEP/quic):", err)
 		return err
 	}
-	defer func() { _ = packetConn.Close() }()
 
-	svc, conn := stun.New(t.cfg, t, packetConn)
+	udpConn, err := net.ListenUDP(network, udpAddr)
+	if err != nil {
+		l.Infoln("Listen (BEP/quic):", err)
+		return err
+	}
+	defer func() { _ = udpConn.Close() }()
+
+	svc, conn := stun.New(t.cfg, t, udpConn)
```

Comment:
I'm really confused about how this thing works. We use pfilter to separate out stun packages from other data. `stun.New` returns the "other data" connection which we call `conn` here and then never use again, except for the deferred close. Instead the `stunQUICWrapper` wraps the underlying UDP conn directly. This should result in two readers of that underlying conn: STUN via pfilter, and QUIC itself. It seems random which packets would end up where...

Reply:
stunQUICWrapper wraps both, udpConn and conn, though there is a problem here that I've noticed here.
We should be using ReadMsgUDP/WriteMsgUDP of the net.PacketConn returned by the stun.New, but we instead circumvent that.

Reply:
I think I fixed it?!

Reply:
I'll step through stuff in the debugger later to day to verify.